From 3e7a2fa8e4e38da2fff631ea0c1bc5f047121e12 Mon Sep 17 00:00:00 2001
From: Matthew Flatt <mflatt@racket-lang.org>
Date: Wed, 4 Apr 2012 16:26:23 -0600
Subject: [PATCH] disable JIT support for vector op on chaprones

---
 src/racket/src/jitcommon.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/racket/src/jitcommon.c b/src/racket/src/jitcommon.c
index f3793d1..61c7197 100644
--- a/src/racket/src/jitcommon.c
+++ b/src/racket/src/jitcommon.c
@@ -1199,6 +1199,7 @@ static int common3(mz_jit_state *jitter, void *_data)
 	}
         if (!ii) {
           __END_SHORT_JUMPS__(1);
+#if 0
           if (iii == 0) {
             if (i)
               (void)mz_beqi_t(sjc.chap_vector_ref_check_index_code, JIT_R0, scheme_chaperone_type, JIT_R2);
@@ -1210,6 +1211,7 @@ static int common3(mz_jit_state *jitter, void *_data)
             else
               (void)mz_beqi_t(sjc.chap_vector_set_code, JIT_R0, scheme_chaperone_type, JIT_R2);
           }
+#endif
           __START_SHORT_JUMPS__(1);
         } else if (ii == -1) {
           /* since we got here, we know that the wrapper is a proxy */
-- 
1.7.4.4

